src/forms/TimeInput.jsx

```diff
@@ -9,26 +9,33 @@ class TimeInput extends React.Component {
 
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: this.props.value || ''
-		};
+
 		this.onChange = this.onChange.bind(this);
 	}
 
+	/**
+	* @function onChange
+	* @param e Event Object
+	* @description called when the input changes, in turn calls the onChange
+	* 	handler prop, if there is one (eg supplied by redux-form) and an onChangeCallback
+	*	which may be provided by a parent component such as DateTimePicker
+	*/
 	onChange(e) {
-		this.setState({ value: e.target.value });
+		this.props.onChange && this.props.onChange(e.target.value);
 		this.props.onChangeCallback && this.props.onChangeCallback(e.target.value);
```

Comment:
I can put `onChange` in the proptypes. `onChange` prop comes from the redux form `Field` component and we want this called if present. `onChangeCallback` is called if the parent component provides it (eg DateTimePicker).
I put an explanation in the header doc.